### PR TITLE
Migrated DualView sample to use the new dual screen SDK version and updated UI tests

### PR DESCRIPTION
**Pull request by bimiron**
- Added the new dual screen dependencies in the build.gradle file
- Created a DualViewApp class to initialize the dual screen components
- Implemented the ScreenInfoListener in both fragments and the activity
- Moved the fragment navigation logic into the activity
- Refactored MapFragment to use the view model instead of the bundle of parcelable data to prevent multiple instances
- Updated UI tests

### CODE REVIEW

File: DualView/src/main/java/com/microsoft/device/display/samples/contentcontext/ListFragment.kt [hunk 1]
```diff
@@ -34,56 +37,56 @@ class ListFragment : Fragment() {
...
+    override fun onResume() {
+        super.onResume()
+        ScreenManagerProvider.getScreenManager().removeScreenInfoListener(this)
     }
 
     private fun onMenuItemSelected(menuItem: MenuItem): Boolean {
         return when (menuItem.itemId) {
             R.id.action_map -> {
-                startDetailsFragment(null)
+                startDetailsFragment(-1)
```
Comment (CesarValiente) [inline DualView/src/main/java/com/microsoft/device/display/samples/contentcontext/ListFragment.kt:89]: Can we please, move this magic number to a constant with a descriptive name? (I know that it was here before, just realized about it now).

File: DualView/src/main/java/com/microsoft/device/display/samples/contentcontext/MapFragment.kt [hunk 2]
```diff
@@ -38,37 +33,54 @@ class MapFragment : Fragment() {
...
-        if (arguments != null) {
-            val item = requireArguments().getParcelable<Restaurant>(Restaurant.KEY)
-            if (item != null && item.mapImageResourceId != 0) {
-                mapView?.setImageResource(item.mapImageResourceId)
-            } else {
-                mapView?.setImageResource(R.drawable.unselected_map)
+        selectedViewModel.selectedPosition.observe(
+            viewLifecycleOwner,
+            {
+                if (it != -1) {
```
Comment (CesarValiente) [inline DualView/src/main/java/com/microsoft/device/display/samples/contentcontext/MapFragment.kt:39]: Same here for this `-1`, better having it in a constant with a descriptive name so we know what's that.

File: DualView/src/main/java/com/microsoft/device/display/samples/contentcontext/view/RestaurantAdapter.kt [hunk 3]
```diff
@@ -34,20 +33,21 @@ class RestaurantAdapter(
...
-
-    fun getItemPosition(item: Restaurant) = items.indexOf(item)
+    override fun getItemCount() = viewModel.listItems.size
 
     fun selectItem(pos: Int) {
-        val oldPosition = selectedPosition
-        selectedPosition = pos
-        notifyItemChanged(selectedPosition)
-        notifyItemChanged(oldPosition)
+        viewModel.selectedPosition.value?.takeIf { it != -1 }?.let {
```
Comment (CesarValiente) [inline DualView/src/main/java/com/microsoft/device/display/samples/contentcontext/view/RestaurantAdapter.kt:46]: Same here

**Review by CesarValiente — APPROVED** [review]
LGTM! Thanks Bianca!
I've just added a few minor comments, for readability, please, fix that and feel free to merge it afterwards.

**Review by ancirja-m — APPROVED** [review]

**Comment by bimiron** [comment]
> LGTM! Thanks Bianca!
> I've just added a few minor comments, for readability, please, fix that and feel free to merge it afterwards.

Thank you Cesar for the comments! :) fixed!

**Pull request merged**